### PR TITLE
register event recorders and add load methods for them

### PR DESCRIPTION
**Pull request by Batch21**

### CODE REVIEW

File: tests/test_recorders.py [hunk 1]
```diff
@@ -1882,6 +1882,16 @@ class TestEventRecorder:
...
 
+    def test_load(self, cyclical_storage_model):
+        """Test load method""" 
+        m = cyclical_storage_model
+        strg = m.nodes['Storage']
+        param = StorageThresholdRecorder(m, strg, 4.0, predicate='<=', name="trigger")
+        EventRecorder.load(m, {"name": "event_rec", "threshold": "trigger", "tracked_parameter":"inpt_flow"})
+        EventDurationRecorder.load(m, {"event_recorder": "event_rec"})
+        EventStatisticRecorder.load(m, {"event_recorder": "event_rec"})
+        m.step()
```
Comment (jetuk) [inline tests/test_recorders.py:1893]: Change this to `m.run()`. I have a feeling something bad happens if you don't specify `recorder_agg_func` or `agg_func`, but it won't happen until the `finish` method.

File: pywr/recorders/events.py [hunk 2]
```diff
@@ -178,6 +178,22 @@ def to_dataframe(self):
 
         return pandas.DataFrame(df_dict)
 
+    @classmethod
+    def load(cls, model, data):
+        from pywr.parameters import load_parameter
+        threshold = data.pop("threshold")
+        try:
+            threshold = load_parameter(model, threshold)
+        except KeyError:
```
Comment (jetuk) [inline pywr/recorders/events.py:187]: I guess we don't test this.

**Review by jetuk — CHANGES_REQUESTED** [review]
Just check that finish method works by changing the test to a full model run.

File: tests/test_recorders.py [hunk 3]
```diff
@@ -1882,6 +1882,17 @@ class TestEventRecorder:
...
     funcs = {"min": np.min, "max": np.max, "mean": np.mean, "median": np.median, "sum": np.sum}
 
+    @pytest.mark.parametrize("threshold_component", [StorageThresholdRecorder, StorageThresholdParameter])
+    def test_load(self, cyclical_storage_model, threshold_component):
+        """Test load method""" 
+        m = cyclical_storage_model
+        strg = m.nodes['Storage']
+        param = threshold_component(m, strg, 4.0, predicate='<=', name="trigger")
+        EventRecorder.load(m, {"name": "event_rec", "threshold": "trigger", "tracked_parameter": "inpt_flow"})
+        EventDurationRecorder.load(m, {"event_recorder": "event_rec", "recorder_agg_func": "max"})
```
Comment (jetuk) [inline tests/test_recorders.py:1892]: So, what happens  if you don't provide `recorder_agg_func` or `agg_func`? Does it error? Should we have a default here?

Comment (Batch21) [reply tests/test_recorders.py:1892]: yeah, it errors. We probably should have a default. Any preferences on what this should be?

Comment (jetuk) [reply tests/test_recorders.py:1892]: I think "mean" is the default for "agg_func" in general, so can likely be the same here. Make sure the docstring says the default value too.

Comment (Batch21) [reply tests/test_recorders.py]: Done

**Review by jetuk — APPROVED** [review]

**Pull request merged**